presidio-image-anonymizer/tests/test_image_anonymizer_api.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+from app import Server
+from presidio_image_anonymizer.entities import InvalidParamException
+
+
+def test_given_no_data_then_we_get_default_dict():
+    assert Server()._get_json_data("") == {}
+
+
+@pytest.mark.parametrize(
+    # fmt: off
+    "str_json",
+    [
+        '{\'color_fill\': \'0, 0, 1\'}',
+        "{'color_fill': '0, 0, 1'}",
+        "{\"color_fill\": \"0, 0, 1\"}",
+    ],
+    # fmt: on
+)
+def test_given_json_string_then_we_get_json_back(str_json):
+    assert Server()._get_json_data(str_json) == {'color_fill': '0, 0, 1'}
+
+
+def test_given_invalid_json_string_then_we_get_an_invalid_param_exception():
+    with pytest.raises(InvalidParamException, match="Invalid json format 'not_json'"):
+        Server()._get_json_data("not_json")
+
+
+def test_given_empty_json_params_then_we_send_default_color_fill():
+    Server()._color_fill_string_to_value({}) == (0, 0, 0)
```

Comment:
You're testing a private method.. and to call it you need the public api endpoint method, which is not suitable for testing here. It's a code smell for moving _color_fill_string_to_value and _get_json_data into different classes, having the service class deal with api related stuff only.

Reply:
There are many debates regarding testing private methods... I prefer having it tested. 
I think moving it to another class is incorrect and kind of detaching it from the context it needs. But I don't mind doing it either.